app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,13 @@
+class RecipesController < ApplicationController
+  def show
+    @recipe = Recipe.find(params[:id])
+  end
+
+  def create_ingredient
+    recipe = Recipe.find(params[:recipe_id])
```

Comment:
A more RESTful here would be to route this action to a joins controller called RecipeIngredientsController. You could move this `create` action to *that* controller. It doesn't matter much for this application, but consider the case where you would want to add functionality here to create a new recipe. You'd probably want to use `recipes#create` for that functionality rather than for adding an ingredient to an existing recipe.